src/main/java/pl/wedding/management/service/user/model/EMail.java

```diff
@@ -0,0 +1,45 @@
+package pl.wedding.management.service.user.model;
+
+import lombok.Getter;
+import org.springframework.util.StringUtils;
+
+import java.util.Objects;
+
+@Getter
+public class EMail {
+
+    private static final String AT_SIGN = "@";
+    private final String email;
+
+    public EMail(String email) {
+
+        if(isInvalid(email)){
+            throw new IllegalArgumentException();
+        }
+        this.email = email;
+    }
+
+    private boolean isInvalid(String value){
+
+        //TODO: Email Validation
+        if(StringUtils.isEmpty(value) || !value.contains(AT_SIGN)){
+            //Current approach is just temporary
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
how about `@EqualsAndHashCode` ?

Reply:
I am not a big fan of many annotations. To use @EqualsAndHasCode in this case, we have to use @EqualsAndHashCode.Include or @EqualsAndHashCode.Exclude on specific field/fields also.